compiler/rustc_middle/src/ty/query.rs

```diff
@@ -106,6 +106,55 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 }
 
+#[inline(always)]
+fn query_get_at<'tcx, Cache, K>(
+    tcx: TyCtxt<'tcx>,
+    execute_query: fn(
+        &'tcx dyn QueryEngine<'tcx>,
+        TyCtxt<'tcx>,
+        Span,
+        Cache::Key,
+        QueryMode,
+    ) -> Option<Cache::Stored>,
+    query_cache: &Cache,
+    span: Span,
+    key: K,
+) -> Cache::Stored
+where
+    K: IntoQueryParam<Cache::Key>,
+    Cache::Stored: Copy,
+    Cache: QueryCache,
+{
+    let key = key.into_query_param();
```

Comment:
Moving this to the caller will make `query_get_at` a bit less generic, hopefully improving compile time.

Reply:
This change was performance neutral and `rustc_driver` size increased by 0.03%, which is not a good sign for compile time.
<table><tr><td rowspan="2">Benchmark</td><td colspan="1"><b>Before</b></th><td colspan="2"><b>After</b></th></tr><tr><td align="right">Time</td><td align="right">Time</td><td align="right">%</th></tr><tr><td>🟣 <b>clap</b>:check</td><td align="right">1.7610s</td><td align="right">1.7607s</td><td align="right"> -0.02%</td></tr><tr><td>🟣 <b>hyper</b>:check</td><td align="right">0.2569s</td><td align="right">0.2570s</td><td align="right"> 0.07%</td></tr><tr><td>🟣 <b>regex</b>:check</td><td align="right">0.9808s</td><td align="right">0.9815s</td><td align="right"> 0.07%</td></tr><tr><td>🟣 <b>syn</b>:check</td><td align="right">1.5819s</td><td align="right">1.5814s</td><td align="right"> -0.03%</td></tr><tr><td>🟣 <b>syntex_syntax</b>:check</td><td align="right">6.1439s</td><td align="right">6.1434s</td><td align="right"> -0.01%</td></tr><tr><td>Total</td><td align="right">10.7246s</td><td align="right">10.7241s</td><td align="right"> -0.00%</td></tr><tr><td>Summary</td><td align="right">1.0000s</td><td align="right">1.0002s</td><td align="right"> 0.02%</td></tr></table>

I've pushed it so we can see what perf says.